src/libs/antares/study/parts/common/series.h

```diff
@@ -33,6 +33,54 @@
 
 namespace Antares::Data
 {
+
+class TimeSeries
+{
+public:
+    using TSNumbers = Matrix<uint32_t>;
+    using TScoefficients = Matrix<double, int32_t>;
+
+    /*!
+     ** \brief Load series from a file
+     **
+     ** \param d Data series
+     ** \param areaID The ID of the area associated to the data series
+     ** \param folder The source folder
+     ** \param filename The filename of the series
+     ** \return A non-zero value if the operation succeeded, 0 otherwise
+     */
+    int timeSeriesLoadFromFolder(Study& s,
+                                 const AreaName& areaID,
+                                 const std::string& folder,
+                                 const std::string& prefix);
+    /*!
+     ** \brief Save time series to a file
+     ** \ingroup windseries
+     **
+     ** \param areaID The ID of the area associated to the data series
+     ** \param folder The target folder
+     ** \param filename The filename of the series
+     ** \return A non-zero value if the operation succeeded, 0 otherwise
+     */
+    int timeSeriesSaveToFolder(const AreaName& areaID, const std::string& folder,
+                               const std::string& prefix) const;
+
+    double getCoefficient(uint32_t year, uint32_t hourInYear) const;
+    const double* getColumn(uint32_t year) const;
+    uint32_t getSeriesIndex(uint32_t year) const;
+
+    void reset();
+
+    bool forceReload(bool reload = false) const;
+    void markAsModified() const;
+    uint64_t memoryUsage() const;
+
+    TScoefficients timeSeries;
+    TSNumbers timeseriesNumbers;
```

Comment:
In some cases (thermal, hydro), we have multiple `TScoefficients` objects sharing a single instance of TSNumbers. For example availability, CO2 cost, fuel cost, etc.

```suggestion
    TScoefficients timeSeries;
    const TSNumbers& timeseriesNumbers;
```